src/test/java/teammates/test/cases/action/AdminActivityLogPageActionTest.java

```diff
@@ -284,95 +283,94 @@ public void statusMessage_validQuery_generatedCorrectly() {
+        verifyLocalTimeInStatusMessage(statusMessage, toDate, Const.SystemParams.ADMIN_TIME_ZONE_ID);
 
         // test statusMessage with `from`
-        query = "from:" + formatterAdminTime.format(yesterday);
+        query = "from:" + formatAdminDate(yesterday);
         action = getAction("filterQuery", query);
-        Calendar fromDate = adminTimeZoneToUtc(getBeginOfTheDayOffsetNowInAdminTimeZone(-1));
+        Instant fromDate = getBeginOfTheDayOffsetNowInAdminTimeZone(-1);
         statusMessage = getShowPageResult(action).getStatusMessage();
-        verifyStatusMessage(statusMessage, 17, 8, fromDate.getTime());
-        verifyLocalTimeInStatusMessage(statusMessage, fromDate.getTime(), Const.SystemParams.ADMIN_TIME_ZONE_DOUBLE);
+        verifyStatusMessage(statusMessage, 17, 8, fromDate);
+        verifyLocalTimeInStatusMessage(statusMessage, fromDate, Const.SystemParams.ADMIN_TIME_ZONE_ID);
     }
 
     @Test(groups = "typicalActivityLogs")
     public void loadingLocalTimeAjaxQuery_validAndInvalidInputs_returnCorrectly() {
-        Calendar now = TimeHelper.now(Const.SystemParams.ADMIN_TIME_ZONE_DOUBLE);
-        SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
+        Instant now = Instant.now();
 
         // Unknown
         String failureMsg = "Local Time Unavailable";
-        verifyLoadingLocalTimeAjaxResult(failureMsg, "Unregistered", "Unknown", now.getTimeInMillis());
-        verifyLoadingLocalTimeAjaxResult(failureMsg, "Instructor", "instructorWithoutCourses", now.getTimeInMillis());
-        verifyLoadingLocalTimeAjaxResult(failureMsg, "Student", "student1InUnregisteredCourse", now.getTimeInMillis());
+        verifyLoadingLocalTimeAjaxResult(failureMsg, "Unregistered", "Unknown", now.toEpochMilli());
+        verifyLoadingLocalTimeAjaxResult(failureMsg, "Instructor", "instructorWithoutCourses", now.toEpochMilli());
+        verifyLoadingLocalTimeAjaxResult(failureMsg, "Student", "student1InUnregisteredCourse", now.toEpochMilli());
         verifyLoadingLocalTimeAjaxResult(failureMsg, "Unregistered:unregisteredCourse",
-                "Unregistered", now.getTimeInMillis());
+                "Unregistered", now.toEpochMilli());
 
         // Role: Admin
-        verifyLoadingLocalTimeAjaxResult(sdf.format(now.getTime()), "Admin", "admin", now.getTimeInMillis());
-        verifyLoadingLocalTimeAjaxResult(sdf.format(now.getTime()), "Student(M)", "admin", now.getTimeInMillis());
+        verifyLoadingLocalTimeAjaxResult(getExpectedAjaxTimeString(now, Const.SystemParams.ADMIN_TIME_ZONE_ID),
+                "Admin", "admin", now.toEpochMilli());
+        verifyLoadingLocalTimeAjaxResult(getExpectedAjaxTimeString(now, Const.SystemParams.ADMIN_TIME_ZONE_ID),
+                "Student(M)", "admin", now.toEpochMilli());
 
         // Role: Instructor
-        verifyLoadingLocalTimeAjaxResult(sdf.format(TimeHelper.convertToUserTimeZone(now, -6).getTime()),
```

Comment:
Not sure where these values of `-6` or `-8` of the old tests are from. Maybe from some other sessions of the same course. Anyway in real life, course should have the same time zone as sessions, so I didn't dig deep into this. I just changed the timezone of the relevant course's time zone in `typicalDataBundle` to match the test.